### ISSUE TITLE
Configuring JURECA with new Usage Model 

### ISSUE THREAD

**Issue by antonelepfl** [cnr-ibf-pa/hbp-bsp-issues#358]
@mmigliore @clupascu As Jureca is getting up and running again, (I think the booster partition is still down) but we need to be prepared, In this ticket probably we can discuss the different things that we have to modify on the code so Jureca can work again using the new Usage Model.

Some things that I have discovered:
The **circuit** was moved from
_/work/pra098/pra09801/CircuitRelease_ to
_/p/project/cpcp0/pcp0068/CircuitRelease/r2017.06b/_

The **home** directory was moved from 
_/homec/vsk25/vsk2512/simulation/_ to
_/p/project/cvsk25/vsk2512_

I'm still trying to find where  _/homec/bp0/bp000022/modulesfiles-x64_ was moved

**Comment by BerndSchuller** [comment]
probably

p/home/bp0/bp000022/jureca/modulesfiles-x64

still figuring this stuff out myself :-)

**Comment by mmigliore** [comment]
Hi Stefano,
please involve Pramod with these issues too, so he can figure out what
happened with this change of accounts at Jureca.
Best
Michele

**Comment by pramodk** [comment]
I was using my old account `vsk2514` and don't think I have access to Jureca. Please email me project details and I can request account.

About the deployment : mostly likely we have re-deploy modules if all directories are moved.

**Comment by BerndSchuller** [comment]
hi,

everyone has only one user ID but multiple projects, so if you have more than one project, you will need to select it using (for example)

X-UNICORE-User-Preferences: group:vsk25

**Comment by clupascu** [comment]
Hi, 

I am not anymore mapped on Jureca under unicore. I get 
Access level at FZJ_JURECA : 'anonymous'
@BerndSchuller Do you have to still remap the users?

**Comment by BerndSchuller** [comment]
we are still not fully up yet, still testing stuff - please give us a bit more time. You'll be mapped to your new user ID.

**Comment by BerndSchuller** [comment]
the mapping is up again - there is still some weirdness with the "bp0*" accounts, I have to check how these should be handled. Carmen you should use your primary account "lupascu1" which has access to everything.

Sorry for the inconvenience, this new usage model is quite a big change for us.

**Comment by clupascu** [comment]
@BerndSchuller Thank you. @alex4200 is continuosly testing the BSP use cases and unicore access was not working anymore on jureca that's why I am asking :)

**Comment by clupascu** [comment]
@BerndSchuller  So as I have u'availableGroups': [u'BP0000', u'CVSK25'],
   u'availableUIDs': [u'bp000028', u'lupascu1'
I have to use primary account 'lupascu1' and 'X-UNICORE-User-Preferences': 'uid:CVSK25' to run on booster partition?

**Comment by BerndSchuller** [comment]
'X-UNICORE-User-Preferences': 'group:cvsk25' 

should be enough - once everything is finished, right now things do not work yet. If you have only a single compute allocation, you do not need any preferences at all.

The "lupascu1" account should have access to the bp00 stuff.

**Comment by mmigliore** [comment]
Hi Pramod,
you should use your new username kumbhar1
You are already listed in the project.
Plese see http://fz-juelich.de/ias/jsc/usage-model for more info on where
the files have been moved.
Let me know of any problem.
Looking forward to know how to launch a sim again...:)
Best
Michele



<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail&utm_term=icon>
Virus-free.
www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail&utm_term=link>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by clupascu** [comment]
@BerndSchuller when I am trying to submit a job on Jureca through Unicore, after I post the job I get "Error submitting job: 403". Any suggestion? Are things supposed to work or not yet?

**Comment by pramodk** [comment]
> Hi Pramod,
you should use your new username kumbhar1
You are already listed in the project.
Plese see http://fz-juelich.de/ias/jsc/usage-model for more info on where
the files have been moved.
Let me know of any problem.
Looking forward to know how to launch a sim again...:)

I have deployed new modules for jureca and booster partition as well. I would like to test with sample simulations but think I am missing something obvious. I have job script as:

```
#!/bin/bash -x

#SBATCH --nodes=2
#SBATCH --ntasks=24
#SBATCH --ntasks-per-node=12
#SBATCH --time=00:20:00
echo "Test"
```
and if I try to submit it as:

```
$ sbatch --account=vsk25 test.sh
sbatch: error: Batch job submission failed: Job violates accounting/QOS policy (job submit limit, user's size and/or time limits)
$ sbatch --account=cvsk25 test.sh
sbatch: error: Batch job submission failed: Invalid account or account/partition combination specified

$ salloc --nodes=1 --account=vsk25 -p devel
salloc: error: Job submit/allocate failed: Job violates accounting/QOS policy (job submit limit, user's size and/or time limits)
```

please let me know.

**Comment by BerndSchuller** [comment]
@clupascu 403 is "forbidden" Are you using an outdated token maybe? Does the account mapping work? i.e. simple GET to the  .../rest/core endpoint ?

**Comment by mmigliore** [comment]
Hi Pramod,

I guess that this is happening because unfortunately we do not have time on
the normal partition and the booster partition is offline for a long
maintenance.
I will ask them to give us more time on the normal partition, but we will
have to wait for booster.
I'll keep you posted.
Best
Michele




<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail&utm_term=icon>
Virus-free.
www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail&utm_term=link>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by BerndSchuller** [comment]
@clupascu  yes, for me it works ... maybe this is yet another issue with the usage model change. I'm currently on a meeting, so apologies it the response times are longer as usual. I'll try to have a look as soon as I can.

**Comment by clupascu** [comment]
@BerndSchuller ok. thanks.

**Comment by clupascu** [comment]
@BerndSchuller in this moment I am 'anonymous' on all the machines on unicore (CINECA-MARCONI, DAINT-CSCS, FZJ_JURECA and BSC-MareNostrum)

**Comment by BerndSchuller** [comment]
@clupascu After an update of the Unity server, the HBP authentication mechanism is completely down. We are on it.

**Comment by antonelepfl** [comment]
Hi Bernd, for Alex and me things are working properly now.

**Comment by antonelepfl** [comment]
I was trying to use booster partition but I get: 
`sbatch: error: Batch job submission failed: Invalid account or account/partition combination specified`
In the scripts that Pramod provided [here](https://github.com/BlueBrain/spack/wiki/JURECA-Deployment#instructions-for-using-software-stack) I see that the partition is `largebooster` but before we were using `booster`. Either way it does not work as expected
I have an sbatch job to try here
`/p/scratch/cvsk25/unicore-jobs/6fc3bab3-f3ce-4f3e-9546-9076ccb68c7d`

As the booster is up again I guess @pramodk you can try again and check

**Comment by mmigliore** [comment]
Booster is ok too. Largebooster was only for large jobs, like the full
circuit. It has not been redefined yet. Smaller jobs can run on booster.

Il giorno Gio 13 Dic 2018, 13:39 Stefano Antonel <notifications@github.com>
ha scritto:

**Comment by clupascu** [comment]
@BerndSchuller I still get the 403 even if I am not using an outdated token.

**Comment by antonelepfl** [comment]
@clupascu if you do GET `https://zam2125.zam.kfa-juelich.de:9112/FZJ_JURECA/rest/core` with the token do you see anything?
I'm able to submit a job thru the GUI and the Notebook.

**Comment by clupascu** [comment]
@antonelepfl Find below the output of GET https://zam2125.zam.kfa-juelich.de:9112/FZJ_JURECA/rest/core

<html>
<head><style type="text/css">table, th, td { border: 1px solid black;  border-collapse: collapse;}</style></head><body>
<h2>Properties</h2>
<table border=1 width='100%'><tr><td>client</td>
<td><ul>
<li><b>role</b>&nbsp;role={selected=user, availableRoles=[Ljava.lang.String;@43d5323d}</li>
<li><b>dn</b>&nbsp;dn=CN=Carmen Alina Lupascu 263955,O=HBP</li>
<li><b>xlogin</b>&nbsp;xlogin={UID=lupascu1, availableGroups=[Ljava.lang.String;@305bacd5, availableUIDs=[Ljava.lang.String;@52628e5f, selectedSupplementaryGroups=null, group=cvsk25}</li>
</ul>
</td>
</tr>
</table>
<h2>Links</h2>
<table border=1 width='100%'><tr><th>Relation</th>
<th>Link</th>
</tr>
<tr><td>registries</td>
<td><a href='https://zam2125.zam.kfa-juelich.de:9112/FZJ_JURECA/rest/core/registries'>https://zam2125.zam.kfa-juelich.de:9112/FZJ_JURECA/rest/core/registries</a></td>
</tr>
<tr><td>jobs</td>
<td><a href='https://zam2125.zam.kfa-juelich.de:9112/FZJ_JURECA/rest/core/jobs'>https://zam2125.zam.kfa-juelich.de:9112/FZJ_JURECA/rest/core/jobs</a></td>
</tr>
<tr><td>sites</td>
<td><a href='https://zam2125.zam.kfa-juelich.de:9112/FZJ_JURECA/rest/core/sites'>https://zam2125.zam.kfa-juelich.de:9112/FZJ_JURECA/rest/core/sites</a></td>
</tr>
<tr><td>storages</td>
<td><a href='https://zam2125.zam.kfa-juelich.de:9112/FZJ_JURECA/rest/core/storages'>https://zam2125.zam.kfa-juelich.de:9112/FZJ_JURECA/rest/core/storages</a></td>
</tr>
<tr><td>transfers</td>
<td><a href='https://zam2125.zam.kfa-juelich.de:9112/FZJ_JURECA/rest/core/transfers'>https://zam2125.zam.kfa-juelich.de:9112/FZJ_JURECA/rest/core/transfers</a></td>
</tr>
<tr><td>factories</td>
<td><a href='https://zam2125.zam.kfa-juelich.de:9112/FZJ_JURECA/rest/core/factories'>https://zam2125.zam.kfa-juelich.de:9112/FZJ_JURECA/rest/core/factories</a></td>
</tr>
<tr><td>storagefactories</td>
<td><a href='https://zam2125.zam.kfa-juelich.de:9112/FZJ_JURECA/rest/core/storagefactories'>https://zam2125.zam.kfa-juelich.de:9112/FZJ_JURECA/rest/core/storagefactories</a></td>
</tr>
</table>
</body></html>                                                                                                                           ✔
──────────────────────────────────────

**Comment by antonelepfl** [comment]
@pramodk I'm trying to run the full scale circuit in Jureca and I'm getting 
`[0] MPI startup(): Imported environment partly inaccesible. Map=0 Info=0` on `stdout`
I'm running the script **input.sh** that contains:
```
module purge all
module load Architecture/KNL
module load Intel
module load IntelMPI
module load HDF5
export MODULEPATH=$MODULEPATH:/p/project/cvsk25/vsk2514/HBP/jureca-booster/install/latest
module load neurodamus/hippocampus
export I_MPI_DEBUG=5
export FULL_CPU_PER_NODE=24
python /p/project/cvsk25/vsk2512/simulation/simulation_launch_no_usertarget.py -vv --blueconfig BlueConfig
```

And **/p/project/cvsk25/vsk2512/simulation/simulation_launch_no_usertarget.py** runs:
```
srun --cpus-per-task=1 --ntasks-per-node=24 --ntasks=24 --nodes 1 /p/project/cvsk25/vsk2514/HBP/jureca-booster/install/10-12-2018/install/linux-centos7-x86_64/intel-19.0.0.117/neurodamus-hippocampus-obgl66/bin/special -NFRAME 1024 /p/project/cvsk25/vsk2514/HBP/jureca-booster/install/10-12-2018/install/linux-centos7-x86_64/intel-19.0.0.117/neurodamus-hippocampus-obgl66/lib/hoclib/init.hoc -mpi
```

The **stderr** contains:
```
New $SOFTWAREROOT: /usr/local/software/jurecabooster
New $LMOD_SYSTEM_NAME: jurecabooster
New $ARCHITECTURE: KNL
Re-setting $MODULEPATH to:
/usr/local/software/jurecabooster/Stages/2018b/UI/Compilers:/usr/local/software/jurecabooster/Stages/2018b/UI/Tools:/usr/local/software/jurecabooster/Stages/2018b/UI/Defaults:/usr/local/software/mod_environment
Was:
/usr/local/software/jurecabooster/Stages/2018b/UI/Compilers:/usr/local/software/jurecabooster/Stages/2018b/UI/Tools:/usr/local/software/jurecabooster/Stages/2018b/UI/Defaults:/usr/local/software/mod_environment

WARNING: Unloading this module might result in a broken environment.
Please swap the module to use the desired architecture instead.

This version of Intel MPI have issues with certain collective functions at large core counts.
If you are affected by such issues and need to use Intel MPI, please switch to the 2018a stage

DEBUG:__main__:Current dir: /p/scratch/cvsk25/unicore-jobs/38ab5251-7748-4e26-9018-0a7ac5926451
DEBUG:__main__:HOC_LIBRARY_PATH: /p/project/cvsk25/vsk2514/HBP/jureca-booster/install/10-12-2018/install/linux-centos7-x86_64/intel-19.0.0.117/neurodamus-hippocampus-obgl66/lib/hoclib
DEBUG:__main__:************** Launching simulation **************
DEBUG:__main__:cmd: ['srun', '--cpus-per-task=1', '--ntasks-per-node=24', '--ntasks=144', '--nodes', '6', '/p/project/cvsk25/vsk2514/HBP/jureca-booster/install/10-12-2018/install/linux-centos7-x86_64/intel-19.0.0.117/neurodamus-hippocampus-obgl66/bin/special', '-NFRAME', '1024', '/p/project/cvsk25/vsk2514/HBP/jureca-booster/install/10-12-2018/install/linux-centos7-x86_64/intel-19.0.0.117/neurodamus-hippocampus-obgl66/lib/hoclib/init.hoc', '-mpi']
```

**Comment by BerndSchuller** [comment]
@clupascu  
I see this error "[ CN=Carmen Alina Lupascu 263955,O=HBP] ... Requested group <CVSK25> is not available." 
Please use the group name as given above, i.e. lower case "cvsk25"

**Comment by clupascu** [comment]
@BerndSchuller Thanks. I can submit jobs now.

**Comment by pramodk** [comment]
> I'm sorry I'm running some slice fo the full circuit. But I have tried with the example that pramod provided and it does not work for me either.

@antonelepfl @mmigliore : I am not able to run standalone mpi benchmark on booster partition (it gets stuck at very beginning). So seems like more system issue to me. I have created ticket to Juelich support : [Ticket#1026363]

**Comment by antonelepfl** [comment]
Hi @pramodk I'm receiving the same issues loading the modules mentioned [here](https://github.com/cnr-ibf-pa/hbp-bsp-issues/issues/358#issuecomment-447279284). Even though we have those warnings the simulation runs and finishes correctly. Do you have any news from the ticket that you open on Juelich?
@clupascu do you receive those warnings mentioned [above](https://github.com/cnr-ibf-pa/hbp-bsp-issues/issues/358#issuecomment-447279284)?

**Comment by pramodk** [comment]
@antonelepfl  : the issue with mpi and deployment is solved...I will forward you the email. See the updated instructions [here](https://github.com/BlueBrain/spack/wiki/JURECA-Deployment#instructions-for-using-software-stack) to use modules.

**Comment by antonelepfl** [comment]
I have used the **booster** partition modules (largobooster is down) and I'm still get the error. I'm running the simulation but you can see it on `/p/scratch/cvsk25/unicore-jobs/9ddbf630-399e-4b1e-bea0-f433c7b40c4d`
I have also created a simple example under that folder/_manual_test_
Do you think it was fixed for largebooster only?

**Comment by antonelepfl** [comment]
The simulations and analysis just finished correctly. Jureca fully operational with simulation.
